File: spring-boot/near/src/main/scala/com/arkr/demos/near/Application.scala
package com.arkr.demos.near

import java.util

import com.arkr.demos.near.config.ApplicationConfig
import org.slf4j.LoggerFactory
import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration
import org.springframework.boot.autoconfigure.{EnableAutoConfiguration, SpringBootApplication}
import org.springframework.boot.{Banner, SpringApplication}
import org.springframework.web.bind.annotation.{RequestMapping, RestController}

/**
  * Created by hztanhuayou on 2017/8/17
  */
@RestController
@SpringBootApplication(scanBasePackageClasses = Array(classOf[ApplicationConfig]))
@EnableAutoConfiguration(exclude = Array(classOf[DataSourceAutoConfiguration]))
class Application {
  private val logger = LoggerFactory.getLogger(classOf[Application])

  private val ut = System.currentTimeMillis()

  @RequestMapping(value = Array("/"))
  def home: Object = {
    logger.info("/ access")
    val myInfo = new util.HashMap[String, Object]()
    myInfo.put("name", "near")
    myInfo.put("version", "3.10")
    myInfo.put("ut", Long.box(ut))
    myInfo
  }

}


object Application {
  private val logger = LoggerFactory.getLogger(classOf[Application])

  def main(args: Array[String]): Unit = {
    val application = new SpringApplication(classOf[Application])
    application.setBannerMode(Banner.Mode.OFF)
    application.run(args: _*)
    logger.info("Started")
  }

}

File: bootstrap/src/main/java/com/arkr/hene/Application.java
package com.arkr.hene;

import com.arkr.hekr.model.User;
import com.arkr.hene.data.dao.UserDAO;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.sql.Date;

/**
 * Created by hztanhuayou on 2017/8/21
 */

@RestController
//@MapperScan(basePackageClasses = DAOInterfaceMaker.class)
@SpringBootApplication
public class Application {
    @Autowired
    private UserDAO userDAO;


    @RequestMapping("/query")
    public User query(@RequestParam("id") Long id) {
        return userDAO.queryById(id);
    }

    @RequestMapping("/insert")
    public Integer insert(@RequestParam("name") String name) {
        User user = new User();
        user.setAge((long) (10000 * Math.random() % 100));
        user.setCtm(new Date(System.currentTimeMillis()));
        user.setUsername(name);
        return userDAO.insert(user);
    }

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }
}
